Rules.py

```diff
@@ -143,65 +163,77 @@ def global_rules(world):
     # I'll have to test how this happens, I haven't played in so long
     # first cycle she gives you magic because you're stuck as deku, afterwards she gives you the mask because you're human
     # right?
     # so what if you start as goron or zora? not sure how that all works
 
+    set_rule(world.get_entrance('Clock Town North Gate'), lambda state: state.form('Human') or state.form('Goron') or state.form('Zora') or state.can('Clock Town Guard Skip'))
 
     ### EAST CLOCK TOWN
-    set_rule(world.get_location('East Clock Town 100 Rupee Chest'), lambda state: state.form('Human') or state.can('Goron Boost') or state.can('Gainer') or state.form('Zora'))
+    set_rule(world.get_location('ECT 100 Rupee Chest'), lambda state: state.form('Human') or state.can('Goron Boost') or state.can('Gainer') or state.form('Zora'))
     # I need to figure out all the actual requirements for this, including tricks, so this one is tentative
     # also the trick names are guesses for sure
 
-    set_rule(world.get_location('Clock Town Maze Minigame HP'), lambda state: state.form('Goron'))
+    set_rule(world.get_location('Treasure Chest Game Goron Prize'), lambda state: state.form('Goron'))
     # oooh I didn't think about it before, do we want to include all the prizes for this?
-    set_rule(world.get_location('Clock Town Maze Minigame 50 Rupees'), lambda state: state.form('Human'))
-    set_rule(world.get_location('Clock Town Maze Minigame 20 Rupees'), lambda state: state.form('Zora'))
-    set_rule(world.get_location('Clock Town Maze Minigame 10 Deku Nuts'), lambda state: state.form('Deku'))
-
-    set_rule(world.get_location('Bomber Hideout 100 Rupee Chest'), lambda state: state.can_blast())
-    # I feel like the goron might be able to just punch this todo: test if goron punch works here
-
-    set_rule(world.get_location('Honey and Darling HP'), lambda state: state.has('Bomb Bag') and state.has('Bow') and state.form('Human'))
-    set_rule(world.get_location('Clock Town Shooting Gallery Quiver Prize'), lambda state: state.has('Bow') and state.form('Human'))
-    set_rule(world.get_location('Clock Town Shooting Gallery HP Prize'), lambda state: state.has('Bow') and state.form('Human'))
+    set_rule(world.get_location('Treasure Chest Game Human Prize'), lambda state: state.form('Human'))
+    set_rule(world.get_location('Treasure Chest Game Zora Prize'), lambda state: state.form('Zora'))
+    set_rule(world.get_location('Treasure Chest Game Deku Prize'), lambda state: state.form('Deku'))
+
+    set_rule(world.get_entrance('Bomber Bouncer'), lambda state: state.has('Bomber Code'))
+    set_rule(world.get_location('Bomber Tunnel Chest'), lambda state: state.can_blast() and (state.form('Human') or state.form('Zora')))
+    set_rule(world.get_entrance('Tunnel Balloon From ECT'), lambda state: state.can_pop_balloon())
+    set_rule(world.get_entrance('Sewers to Tunnel'), lambda state: state.any_form_but('Goron'))
```

Comment:
Yeah apparently goron link is fine in the water because it's not too deep, but isn't able to climb that ledge lol.
I think it was trevorsdiscordaccount or someone else in the testing channel, but they said they'd checked.